app/src/main/java/com/emojimixer/functions/Utils.java

```diff
@@ -37,5 +62,80 @@ public static int getRecyclerCurrentItem(RecyclerView recyclerView, SnapHelper s
         }
     }
 
+    public static void saveImage(ImageView imageView, Context context, String unicodeRep, boolean exportTelegram) {
```

Comment:
Instead of digging into more mess, you could've saved the emoji directly from its URL.